File: skip-the-dishes-query-event-handler/src/main/java/br/com/skip/the/dishes/query/event/handler/commons/QueryEventHandlerConfig.java
package br.com.skip.the.dishes.query.event.handler.commons;

import feign.Feign;
import feign.Logger;
import feign.okhttp.OkHttpClient;
import feign.slf4j.Slf4jLogger;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

@ComponentScan(basePackages = "br.com.skip.the.dishes.query.event.handler")
@Configuration
public class QueryEventHandlerConfig {

    @Value("${event.store.url}")
    private String eventStoreUrl;

    @Bean
    public EventStoreApiService eventStoreApiService(EventStoreBasicAuth eventStoreBasicAuth) {
        return Feign.builder()
                .client(new OkHttpClient())
                .requestInterceptor(eventStoreBasicAuth)
                .logger(new Slf4jLogger())
                .logLevel(Logger.Level.FULL)
                .target(EventStoreApiService.class, eventStoreUrl);
    }

}

File: skip-the-dishes-query-event-handler/src/main/kotlin/br/com/skip/the/dishes/query/event/handler/commons/QueryEventHandlerConfig.kt
package br.com.skip.the.dishes.query.event.handler.commons

import feign.Feign
import feign.Logger
import feign.okhttp.OkHttpClient
import feign.slf4j.Slf4jLogger
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@ComponentScan(basePackages = ["br.com.skip.the.dishes.query.event.handler"])
@Configuration
class QueryEventHandlerConfig(@Value("\${event.store.url}")
                              private val eventStoreUrl: String) {

    @Bean
    fun eventStoreApiService(eventStoreBasicAuth: EventStoreBasicAuth): EventStoreApiService =
            Feign.builder()
                    .client(OkHttpClient())
                    .requestInterceptor(eventStoreBasicAuth)
                    .logger(Slf4jLogger())
                    .logLevel(Logger.Level.FULL)
                    .target(EventStoreApiService::class.java, eventStoreUrl)

}
